sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -232,16 +232,28 @@ Command *Scheduler::GraphBuilder::insertMemoryMove(MemObjRecord *Record,
 
   AllocaCommandBase *AllocaCmdSrc =
       findAllocaForReq(Record, Req, Record->MCurContext);
+  if (!AllocaCmdSrc && IsSuitableSubReq(Req)) {
+    auto IsSuitableAlloca = [Record, Req](AllocaCommandBase *AllocaCmd) {
+      bool Res = sameCtx(AllocaCmd->getQueue()->get_context_impl(),
+                         Record->MCurContext) &&
+                 AllocaCmd->getRequirement()->MSYCLMemObj == Req->MSYCLMemObj &&
+                 AllocaCmd->getType() == Command::CommandType::ALLOCA;
```

Comment:
What do you think about modifying findAllocaForReq so if finds "best matching" alloca?